### ISSUE TITLE
Cannot add attachments

### ISSUE THREAD

**Issue by lmasellis** [thunderbird/thunderbird-android#7423]
### Checklist

- [X] I have used the search function to see if someone else has already submitted the same bug report.
- [X] I will describe the problem with as much detail as possible.

### App version

6.603

### Where did you get the app from?

F-Droid

### Android version

Android 13

### Steps to reproduce

1. Open a received email
2. Click on reply icon
3. Click on attachment icon
4. Select a file

### Expected behavior

Attachment is visible at the bottom of the message.

### Actual behavior

No attachment was actually added to the email.

~~Note: adding attachments works without issues when working on new messages (not replies).~~

Edit: checked again and actually the same behavior is experienced also when composing new messages from scratch. The only way I can now send an attachment is by sharing a file from some other app (e.g. a file manager) to K-9. This issues seems very recent (weeks?), though cannot confirm exactly when appeared first.

**Comment by lmasellis** [comment]
The bug is still present as of now, with K-9 ver 8.2 from F-droid, Android 14 on Samsung Galaxy A33 5G.

When I hit the paperclip botton, the only action proposed is "media selector"; if I choose this action, I can select the file to be attached, but then the file is not actually attached to the email.

**Comment by kewisch** [comment]
7373 covers both the file and camera cases and we have a design there. Thanks for filing!

**Comment by lmasellis** [comment]
I do not really agree this is a duplicate. 7323 is about a new feature, this. bug is instead about an existing feature (not a minor one, by the way!) that doesn't work!

**Comment by kewisch** [comment]
Ah it does look like I misunderstood. So you are saying that you can't attach files at all, other than sharing with the app? I remember we did a few changes in this area. Could you maybe try 8.0 or the last version before to see if it acts the same way? For 8.0 you can also test Thunderbird for Android as it would be working off the same codebase, in case you don't have a separate profile or device available to test.

**Comment by lmasellis** [comment]
Yes, the problem is that I can select the file I want to attach, but after it is selected I get back to the message being composed and no attachment shows up.
I tried Thunderbird 8.0 from F-droid but everything is the same. By the way it was the same back with 6.603, as per initial report.
It also seems that this behavior depends on the source of the file, e.g.:
- files downloaded to the phone from the the web through Firefox are NOT attached correctly
- files downloaded to the phone via Bluetooth are NOT attached correctly
- photos shot by the camera app are attached correctly
- files received via email as attachments and saved from K-9 to the phone storage are attached correctly
- files selected directly from Google Drive are attached correctly
All in all, it seems that there is some permission problem, so that K-9 is able to access files "owned" by some apps, but not the ones "owned" by some other app.

**Comment by cfitz25** [comment]
any updates on this? ive just ran into it now

**Comment by ghpille** [comment]
I can't attach a file when composing a message. Nothing seems to happen when I choose "Attach File".  I succeed in attaching a picture via the camera app, but the app almost freezes when this is done.  After a couple of minutes it becomes responsive again.  I don't seem to remember K-9 mail ever asking permission to access files on the phone.  I'm running version 11.1 on Android 13.  I can share a file, one file at a time, via K-9 mail.

**Comment by coreycb** [comment]
I just tested on 13.0b4 and was able to successfully send an email with an attachment. Would you be interested in giving beta a try?

**Comment by ghpille** [comment]
Was that something you couldn't do with K-9?  I learned how to install an unknown app on Android 13, but to no avail.  Attach file freezes Thunderbird Beta.  Even better, it makes the phone reboot!

**Comment by holzkohlengrill** [comment]
I experience this issue also for a couple of weeks.
However this happens only for specific email conversations. I couldn't figure out what's the criteria.

Also when replying (text was entered) when selecting an attachment (or not choose but instead try to go back to the draft) the drafted message is gone and I am back in reading mode. 

I tired clearing the old message text,changed the subject without success. So it seems not related to illegal characters or character sequenced.
